### PR TITLE
implement `--invert` for `pip tree`

### PR DESCRIPTION
**Pull request by ChannyClaus**
## Summary
partially resolves https://github.com/astral-sh/uv/issues/4439

## Test Plan
existing tests pass + added a couple of new tests with `--invert`.

### CODE REVIEW

File: crates/uv/src/commands/pip/tree.rs [hunk 1]
```diff
@@ -187,19 +198,20 @@ impl<'a> DisplayDependencyGraph<'a> {
...
         let mut lines = vec![line];
-
+        let empty_vec = Vec::new();
         path.push(package_name.clone());
         visited.insert(package_name.clone());
-        let required_packages = required_with_no_extra(installed_dist, self.markers);
+        let required_packages = self
+            .requires_map
+            .get(installed_dist.name())
+            .unwrap_or(&empty_vec)
```
Comment (ibraheemdev) [inline crates/uv/src/commands/pip/tree.rs:207]: You can use `unwrap_or_default` here.

File: crates/uv/tests/pip_tree.rs [hunk 2]
```diff
@@ -170,6 +170,55 @@ fn nested_dependencies() {
...
+    Installed 5 packages in [TIME]
+     + joblib==1.3.2
+     + numpy==1.26.4
+     + scikit-learn==1.4.1.post1
+     + scipy==1.12.0
+     + threadpoolctl==3.4.0
+    "###
+    );
+
+    uv_snapshot!(context.filters(), tree_command(&context).arg("--invert"), @r###"
```
Comment (ibraheemdev) [inline crates/uv/tests/pip_tree.rs:203]: Can you add the non-inverted output to these tests too so I can see the difference?

Comment (ChannyClaus) [reply crates/uv/tests/pip_tree.rs:203]: the new test is exactly the same as the existing test https://github.com/astral-sh/uv/blob/main/crates/uv/tests/pip_tree.rs#L127-L171 with the only difference `--invert` (same for the other one)

File: crates/uv/src/commands/pip/tree.rs (hunk 1 above)
Comment (ChannyClaus) [reply crates/uv/src/commands/pip/tree.rs:207]: seems like this gives 
```
error[E0277]: the trait bound `&Vec<PackageName>: std::default::Default` is not satisfied
    --> crates/uv/src/commands/pip/tree.rs:206:14
     |
206  |             .unwrap_or_default()
     |              ^^^^^^^^^^^^^^^^^ the trait `std::default::Default` is not implemented for `&Vec<PackageName>`
```
maybe i'm using it wrong?

Comment (ibraheemdev) [reply crates/uv/src/commands/pip/tree.rs:207]: Ah, sorry. You could use `.iter().flatten()` instead, but this is fine.

**Review by ibraheemdev — COMMENTED** [review]
Can you add a test to see how `--invert` interacts with `--no-dedupe`? After that this looks good to me.

**Comment by ChannyClaus** [comment]
all done 🌵

**Comment by Peiffap** [comment]
I haven't looked at the implementation in detail, but is there a reason we want this to be `--invert` instead of `--reverse` like in `pipdeptree`?

**Comment by ChannyClaus** [comment]
> I haven't looked at the implementation in detail, but is there a reason we want this to be `--invert` instead of `--reverse` like in `pipdeptree`?

i followed the naming convention in `cargo tree`, but if there's an overall preference for `--reverse` i'm happy to rename 👍

**Comment by zanieb** [comment]
I kind of find `--invert` clearer, but we can provide a hidden alias `--reverse` for pipdeptree compatibility.

**Review by ibraheemdev — APPROVED** [review]

**Pull request merged**